hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1845,12 +1851,22 @@ public void move(final byte[] encodedRegionName,
       // closed
       serverManager.sendRegionWarmup(rp.getDestination(), hri);
 
+      // Here wait until all the meta regions are not in transition.
+      if (!hri.isMetaRegion() && assignmentManager.getRegionStates().isMetaRegionInTransition()) {
+        Thread.sleep(timeoutWaitMetaRegionAssignment);
+        if (assignmentManager.getRegionStates().isMetaRegionInTransition()) {
+          LOG.info("Moving " + rp + " failed. " +
+            "While there is still meta regions in transition after " +
+            timeoutWaitMetaRegionAssignment + "ms waiting.");
```

Comment:
I think ERROR log would be better. Also, how about this error message?

`LOG.error("This is fail-fast of the region move because hbase:meta region is in transition. Failed region move info: " + rp);`

Reply:
Looks better. Changed in the latest commit.